src/securityPolicy.c

```diff
@@ -370,23 +449,50 @@ security_policy_t policyForSignTxOutputAddressParams(
 		break;
 	}
 
+	// this captures the essence of a change output: money stays
+	// on an address where payment is fully controlled by this device
+	DENY_UNLESS(determineSpendingChoice(params->type) == SPENDING_PATH);
+	// Note: if we allowed script hash in spending part, we must add a warning
+	// for missing datum (see policyForSignTxOutputAddressBytes)
+
+	ASSERT(determineSpendingChoice(params->type) == SPENDING_PATH);
+	DENY_IF(violatesSingleAccountOrStoreIt(&params->spendingKeyPath));
+
+	if (includeDatumHash) {
+		// together with the above requirement on SPENDING_PATH,
+		// this forbids datum in change outputs entirely
+		DENY_UNLESS(allows_datum_hash(params->type));
+
+		// no Plutus elements for pool registration
+		DENY_IF(txSigningMode == SIGN_TX_SIGNINGMODE_POOL_REGISTRATION_OWNER);
+		DENY_IF(txSigningMode == SIGN_TX_SIGNINGMODE_POOL_REGISTRATION_OPERATOR);
+	}
+
 	switch (txSigningMode) {
 
 	case SIGN_TX_SIGNINGMODE_POOL_REGISTRATION_OPERATOR:
 	case SIGN_TX_SIGNINGMODE_ORDINARY_TX: {
-		DENY_UNLESS(determineSpendingChoice(params->type) == SPENDING_PATH);
-		DENY_IF(violatesSingleAccountOrStoreIt(&params->spendingKeyPath));
 		SHOW_UNLESS(is_standard_base_address(params));
+		SHOW_IF(includeDatumHash);
 		ALLOW();
 		break;
 	}
 
 	case SIGN_TX_SIGNINGMODE_MULTISIG_TX: {
 		// all outputs should be given as external addresses
+		// because generally, more than one party is needed to sign
+		// spending from a multisig address, so no change outputs
 		DENY();
 		break;
 	}
 
+	case SIGN_TX_SIGNINGMODE_PLUTUS_TX: {
+		// the output could affect script validation so it must not be entirely hidden
+		// Note: if we relax this, some of the above restrictions may apply
+		SHOW();
+		break;
+	}
```

Comment:
Just to make sure this is a conscious decision - do we really want to call the device-owned outputs in Plutus transactions "change" (since it's not technically a change)?

Reply:
I think "change output" has some tradition and this more or less corresponds to the definition 
https://cips.cardano.org/cips/cip2/#changeoutput
even for Plutus. [It is hard to precisely define what change is, e.g. when part of the output comes straight from the input exclusively cotrolled by the HW wallet and another part from some script-controlled stuff. Also, one might want to write a script that controls his spending from his HW wallet, and then fund the script address from the wallet, in which case we are totally talking about "change" when spending the script-controlled money, at least in the economic sense.]

I am quite sure "device-owned" is more confusing for the users. I'd be willing to consider other options, but I am not aware of anything better than "change output".